tests/Owin.Security.OpenIdConnect.Server.Tests/OpenidConnectTestServer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Owin;
+using Microsoft.Owin.Testing;
+using Newtonsoft.Json.Linq;
+
+namespace Owin.Security.OpenIdConnect.Server.Tests
+{
+    public class OpenIdConnectTestServer : TestServer
+    {
+        public OpenIdConnectTestServer(Action<OpenIdConnectTestServer> configure = null)
+        {
+            var clock = new TestClock();
+            Options = new OpenIdConnectServerOptions
+            {
+                SystemClock = clock,
+                AuthorizationEndpointPath = new PathString("/authorize"),
+                TokenEndpointPath = new PathString("/token"),
+            };
+
+            if (configure != null)
+            {
+                configure(this);
+            }
+
+            Configure(app =>
+            {
+                app.Properties["host.AppName"] = "Owin.Security.OpenIdConnect.Server.Tests";
+                app.UseOpenIdConnectServer(Options);
+            });
+        }
+
+        public async Task<Transaction> SendAsync(string uri, string postBody = null)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, uri);
+            if (!string.IsNullOrEmpty(postBody))
+            {
+                request.Method = HttpMethod.Post;
+                request.Content = new StringContent(postBody, Encoding.UTF8, "application/x-www-form-urlencoded");
+            }
+
+            var transaction = new Transaction
+            {
+                Request = request,
+                Response = await HttpClient.SendAsync(request),
+            };
+
+            transaction.ResponseText = await transaction.Response.Content.ReadAsStringAsync();
+
+            if (transaction.Response.Content != null &&
+                transaction.Response.Content.Headers.ContentType != null &&
+                transaction.Response.Content.Headers.ContentType.MediaType == "application/json")
+            {
+                transaction.ResponseToken = JToken.Parse(transaction.ResponseText);
+            }
+
+            return transaction;
+        }
+
+        public OpenIdConnectServerOptions Options { get; set; }
+
+        public class Transaction
```

Comment:
If `HttpResponseMessage.Content.ReadAsString()` is not convenient, we should indeed consider adding new extensions for `HttpResponseMessage` and `HttpContext`.

These extensions would probably be good candidates for a new `aspnet-contrib` shared project :smile: